lib/statement.rb

```diff
@@ -0,0 +1,38 @@
+class Statement 
+
+  attr_reader :tracker
+
+  def initialize(tracker = Tracker.new)
+    @tracker = tracker
+  end
+
+  def compose_line(date, credit, debit, balance)
```

Comment:
Really nice method that's focused on one responsibility! One handy trick to avoid the `line[1]` stuff:

```
[date,
 credit.nil? ? "" : credit,
 debit.nil? ? "" : debit,
 balance].join(" ||  ")
```